beir/retrieval/search/dense/exact_search.py

```diff
@@ -45,6 +46,9 @@ def search(self,
         logger.info("Sorting Corpus by document length (Longest first)...")
 
         corpus_ids = sorted(corpus, key=lambda k: len(corpus[k].get("title", "") + corpus[k].get("text", "")), reverse=True)
+        if ignore_identical_ids:
+            # We remove the query from results if it exists in corpus
+            corpus_ids = [cid for cid in corpus_ids if cid not in query_ids] 
```

Comment:
Doesn't this make the task "easier" by removing all other queries as options for each query?

I.e. previously, given query1 the model could wrongly retrieve query2 (if it was also in the corpus). 
Now the model cannot retrieve any of the other queries which makes it easier assuming the answer is never another query.

Reply:
I think thus option was for Quora: You want to find paraphrases of queries, but not the original start query. But this original query will always be ranked first at it is also part of the corpus

Reply:
Which is why we have the `ignore_identical_ids` option I think. This PR only tries to fix `ignore_identical_ids=True` case